ddht/v5_1/client.py

```diff
@@ -437,3 +463,126 @@ async def topic_query(
         request_id: Optional[bytes] = None,
     ) -> InboundMessage[FoundNodesMessage]:
         raise NotImplementedError
+
+
+@asynccontextmanager
+async def common_client_stream_find_nodes(
```

Comment:
This isn't actually "common" yet - I tried to implement it on the Alexandria client, but since an Alexandria client doesn't have a `dispatcher`, it wasn't a straightforward integration so I decided to leave it for a following pr if we want this api on the alexandria client/network

Reply:
I think you're right to do it in a separate PR.  Think it's going to be a little complicated to do it cleanly because we *can* expose a common `subscribe(...)` API but they use different message types for their `FindNodes` messages which makes a common implementation a bit more complex but doable.

First step is going to be proxying the subscription API so that it can be accessed directly from `NetworkAPI.subscribe(...)` (which under the hood simply proxies to the dispatcher).

Next you'll have to adjust this function to take the type of the `FindNodesMessage`.

And *then* you'll still have to deal with the fact that the messages are shaped differently and so you won't be able to access their payloads the same way... icky.